From dd5afdb497e5e8d9633dfeeae573a476ea7a404e Mon Sep 17 00:00:00 2001
From: Jacky Bai <ping.bai@nxp.com>
Date: Fri, 12 Jul 2019 15:05:51 +0800
Subject: [PATCH 1/9] MLK-22207 plat: imx8m: Fix the data sync issue in dram
 dvfs flow

Explict memory barrier(DSB) is necessary to make sure
other cores observe the correct flags updated by the primary
core before the primary begins doing DRAM DVFS.

Signed-off-by: Jacky Bai <ping.bai@nxp.com>
(cherry picked from commit dac8d677447c153cd148fee9e35445c24162f190)
---
 plat/imx/common/imx8m/dram.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/imx/common/imx8m/dram.c b/plat/imx/common/imx8m/dram.c
index de58cf2cc..d567cca37 100644
--- a/plat/imx/common/imx8m/dram.c
+++ b/plat/imx/common/imx8m/dram.c
@@ -228,6 +228,7 @@ int dram_dvfs_handler(uint32_t smc_fid,
 		/* set the WFE done status */
 		spin_lock(&dfs_lock);
 		wfe_done |= (1 << cpu_id * 8);
+		dsb();
 		spin_unlock(&dfs_lock);
 
 		while (1) {
@@ -239,6 +240,7 @@ int dram_dvfs_handler(uint32_t smc_fid,
 		}
 	} else {
 		wait_ddrc_hwffc_done = true;
+		dsb();
 		/* trigger the IRQ */
 		for (int i = 0; i < 4; i++) {
 			int irq = irqs_used[i] % 32;
-- 
2.17.1

